modules/ctl/README.md

```diff
@@ -0,0 +1,197 @@
+
+#### Packaging the Application
+
+In order to deply the app we must first create a **docker image** of our application. We do this via sbt-native-packager as follows:
+
+```
+> telnetd/docker:publishLocal
+```
+
+Image names have the form `<repo>/<artifact>:<version>`. Ours will be `geminihlsw/gem-telnetd:<version>` where the version is the git commit hash, or the hash followed by `-UNCOMMITTED` if there are uncommitted changes. This give us very fine-grained versioning and a reliable partial ordering to know that an upgrade is legal. Note that if the filesystem version isn't the same as the version computed by sbt you'll need to reload the project. The prompt will include a message to this effect.
+
+The image will be stored in docker's internal storage area, which shares "slices" with other images built from the same base to save space.
+
+For remote deployments we must publish and push the image to a repository where it can be seen by other machines. For now we're using docker-hub but it probably makes sense to set up an internal repository. To do this you need to create a docker-hub account and be added to the `geminihlsw` organization (talk to Rob), and then you can do `telnetd/docker:publish` to push the image up to the cloud. Note that you need to do this in order to perform remote deployments, even if they're to a VM on your own machine.
+
+#### Initial Deployment
+
+Deployments can be upgrades (the default) or standalone. Since we have no deployment to upgrade yet we'll do a standalone deployment. At the sbt prompt do:
+
+```
+> gemctl deploy -s
+```
+
+By default the deployment will use the current git HEAD as the version, and will deploy locally. Lots of things will happen and some things will be downloaded from the internet, and eventually you should end up with a running application. The log is pretty informative and shows the various things that are being done. If you run with `-v` you will see all the commands that are being issued.
+
+
+```
+> gemctl deploy
+[info] Updating {file:/Users/rnorris/Scala/gem/}ctl...
+[info] Resolving jline#jline;2.12.1 ...
+[info] Done updating.
+[info] Running gem.ctl.main deploy
+
+[info]  Target host is localhost
+[info]  Deploy number for this host will be 3.
+[info]  Verifying deploy commit.
+[info]    Using HEAD.
+...
+[info]    Deploying Gem.
+[info]      Creating Gem container from image 31f970aef7c7
+[info]        Container is gem-3 8a6a3e54ef47ba3e79ce1b473939d74cd83c1c786d137a2be30d54cb226d872f.
+[info]      Service is available at localhost:1234.
+
+[success] Total time: 18 s, completed Mar 30, 2017 4:09:32 PM
+>
+```
+
+Log in as root with no password, then change the password just to commit a change to the database.
```

Comment:
Sure, we could do that. This is the root user on Gem (not the server) just to be clear. In any real deployment the root password hash will be copied over with the rest of the application data, so this is really just an issue for initial deployments.